applications/tari_base_node/src/parser.rs

```diff
@@ -447,6 +457,23 @@ impl Parser {
         println!("Version: {}", VERSION);
     }
 
+    /// Function to process the ge-state-info command
+    fn process_state_info(&mut self) {
+        let mut channel = self.state_machine_info.clone();
+        self.executor.spawn(async move {
+            match channel.next().await {
```

Comment:
This would not necessarily have the current state of the base node, or any state change queued up - which would make the command look as if it doesn't do anything - what we really need is a handle function that can query the base node status.

Reply:
I covered all the normal states in the channel It should have the state. 
I think the only states not covered, is startup and closing.

Reply:
Thanks for the comments - yup I see how it works now :)